Examples/MigrateToNewIfLetSyntax.swift

```diff
@@ -0,0 +1,59 @@
+import Foundation
+import SwiftSyntax
+import SwiftParser
+
+/// MigrateToNewIfLetSyntax will visit each if statement in the Syntax tree, and
+/// checks if the there is an if condition which is of the pre Swift 5.7 "if-let-style"
+/// and rewrites it to the new one
+///
+/// For example will it turn:
+/// ```
+/// if let foo = foo {
+///   ...
+/// }
+/// ```
+/// into:
+/// ```
+/// if let foo {
+///   ...
+/// }
+class MigrateToNewIfLetSyntax: SyntaxRewriter {
+  // Visit over all IfStmtSyntax nodes
+  override func visit(_ node: IfStmtSyntax) -> StmtSyntax {
```

Comment:
Only `if` statements are currently considered, but optional bindings are allowed in `guard` and `while` statements as well. For an example, it might be fine to cover only one case and leave the rest up to the motivated user, though. 😉

Shameless plug by the way: There already is such a [rule](https://github.com/realm/SwiftLint/blob/main/Source/SwiftLintFramework/Rules/Idiomatic/IfLetShadowingRule.swift) available in [SwiftLint](https://github.com/realm/SwiftLint).

Reply:
> For an example, it might be fine to cover only one case and leave the rest up to the motivated user, though.

That was exactly my thought 😉